crypto/eth2_hashing/src/lib.rs

```diff
@@ -127,16 +127,15 @@ pub enum DynamicImpl {
 // Runtime latch for detecting the availability of SHA extensions on x86_64.
 //
 // Inspired by the runtime switch within the `sha2` crate itself.
-#[cfg(target_arch = "x86_64")]
+#[cfg(all(feature = "detect-cpufeatures", target_arch = "x86_64"))]
 cpufeatures::new!(x86_sha_extensions, "sha", "sse2", "ssse3", "sse4.1");
 
 #[inline(always)]
 pub fn have_sha_extensions() -> bool {
-    #[cfg(target_arch = "x86_64")]
+    #[cfg(all(feature = "detect-cpufeatures", target_arch = "x86_64"))]
     return x86_sha_extensions::get();
 
-    #[cfg(not(target_arch = "x86_64"))]
```

Comment:
We need to keep a `cfg` here to prevent the compiler from warning about unreachable code

Reply:
we want the negation of the other condition: `cfg(not(all(feature = "detect-cpufeatures", target_arch = "x86_64")))`

Reply:
Yeah, that makes sense! Fixed in be49f5b